### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by nmattia**
Adds a [CircleCI] config that builds the survey and uploads the build artifacts to cachix. Next steps:

* Login to CircleCI and enable the project
* Add your cachix token to the env
* Add the badge to the README (coz it's cool)

[CircleCI]: https://circleci.com

CC @nh2 @basvandijk

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      - run:
+          name: Install cachix
+          command: |
+            nix-env -iA cachix -f https://cachix.org/api/v1/install
+
+      - run:
+          name: Run cachix
+          command: |
+            cachix push deckgo --watch-store
```
Comment (nh2) [inline .circleci/config.yml]: Should this be `static-haskell-nix`?

Don't we have to authenticate it somehow so that it can upload?

**Comment by nh2** [comment]
Some more questions:

* Building GHCs + all of Stackage can take very long (many hours on a performant workstation). Can CircleCI handle that, or will builds take forever? Last time I tried it, its VMs were very very slow.
* Is there a maximum duration for builds?
* Is it possible to build on dedicated servers as build slaves, e.g. to run on a cheap Hetzner?

File: .circleci/config.yml (hunk 1 above)
Comment (nmattia) [reply .circleci/config.yml]: Definitely should be static-haskell-nix, copy paste error. The authentication happens through the token passed as an env var, See commit message

Comment (nh2) [reply .circleci/config.yml]: Makes sense, thanks!

**Comment by nh2** [comment]
Now suddently I'm getting:

https://circleci.com/gh/nh2/static-haskell-nix/4

```
Fatal error: glibc detected an invalid stdio handle
Aborted (core dumped)
Exited with code 134
```

**Comment by nmattia** [comment]
Workaround: https://github.com/nmattia/niv/blob/ed8c23faeff2fd3c54677b8ef381e7baad8280f2/.circleci/config.yml#L35

**Comment by nh2** [comment]
Hmm OK. Trying with https://github.com/nh2/static-haskell-nix/commit/bc4b14493bd6f9af4a64dabbbf470e3c0c5897e5

https://circleci.com/gh/nh2/static-haskell-nix/5

**Comment by nh2** [comment]
@nmattia Hmm getting the next weird stuff:

https://circleci.com/gh/nh2/static-haskell-nix/5

```
#!/bin/bash -eo pipefail
cachix push static-haskell-nix --watch-store

Watching /nix/store for new builds ...
pushing /nix/store/y6kzzrm6lw5f9f2cpwv5cshg1kmpy28d-hedgehog-0.6.1
...
pushing /nix/store/i4l84xrwpskjxsaw3h5k1jygxvmdjq73-servant-client-0.15
pushing /nix/store/fc6mxy4wg6byr2m1far14qyh32df8fnm-servant-swagger-ui-core-0.3.2
pushing /nix/store/26z0kipp9am1fgcw8r3vf0qlmcnh7syi-servant-conduit-0.15
pushing /nix/store/1rpk3hkwphnq0jbwjnvhj0dibv67bs6s-cachix-api-0.2.0
Step was canceled
```

`Step was canceled`?

Also it still went green nevertheless, but that stage went grey somehow.

**Comment by nmattia** [comment]
That's normal; cachix runs in the background and gets canceled when the main job finishes

**Comment by nh2** [comment]
@nmattia ~Ah, is this perhaps because it runs as a daemon?~ I should have refreshed, seeing your answer now.

If yes, it it possble to make that clear somehow?

**Comment by nmattia** [comment]
Make it clearer from what point of view? It's currently using CircleCI's support for [background](https://github.com/nh2/static-haskell-nix/blob/41a47d228f8048ac7437fb6e4201415897c40a37/.circleci/config.yml#L44) steps but you could use `cachix ... &` instead

**Comment by nh2** [comment]
I'll just rename the section heading to `Run cachix in background (it is normal for this to be cancelled at the end)`, that should do.

Cool this is working now!

**Pull request merged**